### PR TITLE
Test env

### PR DESCRIPTION
**Pull request by scott-wyatt**

### CODE REVIEW

File: src/app/ng-intercom/intercom/intercom.ts [hunk 1]
```diff
@@ -43,17 +44,16 @@ export class Intercom {
...
-   * @param {object} [intercomData] Your intercom configuration
    */
-  boot(intercomData?: BootInput) {
+  public boot(intercomData?: BootInput): void {
     if (!isPlatformBrowser(this.platformId)) {
       return
     }
 
-    const data = {
+    const data = Object.assign({}, {
```
Comment (wbhob) [inline src/app/ng-intercom/intercom/intercom.ts]: This seems like a more verbose way to write what it already did. Is this Object.assign necessary?

File: src/app/ng-intercom/types/boot-input.ts [hunk 2]
```diff
@@ -14,5 +14,5 @@ export interface DefaultBootInput {
     custom_launcher_selector?: string;
 }
 
-export type BootInput = ((UserWithEmail | UserWithUid) & DefaultBootInput & Any) | DefaultBootInput;
+export type BootInput = ((UserWithEmail | UserWithUid) & DefaultBootInput & Any) | DefaultBootInput
```
Comment (wbhob) [inline src/app/ng-intercom/types/boot-input.ts:17]: Also, if you know a better way to do this than this I would love it, since this is very verbose.

**Review by wbhob — COMMENTED** [review]
Overall good, just a couple of questions.

**Comment by scott-wyatt** [comment]
So these test are failing because to get this to work with Yarn I had to remove the node engine restriction. It's hard to test that this will work as a library when it's not published to NPM, but this should fix at least one open issue and hopefully the AOT stuff.

**Comment by scott-wyatt** [comment]
It's just so strange, because it passes AOT and SSR in the package, but then when it's published, AOT can throw a complaint about the library `Only initialized variables and constants can be referenced in decorators because the value of this variable is needed by the template compiler in` config.  Tracking this down it looks like it hates the way the interface class is exported. But hopefully this was fixed in the latest version of ng-packagr

**Comment by scott-wyatt** [comment]
Confirmed, this works with AOT and SSR now. It was an undocumented ng-packagr error were using an index barrel computed multiple result paths. Merging and publishing.

**Pull request merged**